src/arch-clap.ts

```diff
@@ -0,0 +1,518 @@
+      console.log("INFO: updated tsconfig.json for you.  Please commit it");
+    }
+  }
+
+  setupPublishingConfig(): void {
+    const files = this.appPkg.files || [];
+
+    if (this.hasTypeScript) {
+      files.push("dist");
+    }
+
+    if (Fs.existsSync(Path.resolve("lib"))) {
+      files.push("lib");
+    }
+
+    this.appPkg.files = _.uniq(files).sort();
+    if (this.saveAppPkgJson()) {
+      console.log(`INFO: updated files in your package.json for publishing.`);
+    }
+  }
+
+  setupCompileScripts(): void {
+    if (!this.hasTypeScript) {
+      return;
+    }
+    const scripts = this.appPkg.scripts || {};
+    this.appPkg.scripts = scripts;
+    _.defaults(scripts, {
+      compile: "tsc",
+      prepublishOnly: "tsc"
+    });
+    if (this.saveAppPkgJson()) {
+      console.log(`INFO: added compile and prepublishOnly npm scripts for your typescript.`);
+    }
+  }
+
+  setupMocha(): void {
+    const mochaOpts = this.appPkg.mocha || {};
+    this.appPkg.mocha = mochaOpts;
+
+    const tsNodeReg = "ts-node/register";
+    const sourceMapReg = "source-map-support/register";
+    const selfPkg = "@xarc/module-dev";
+    const withSelf = this.appHasDevDeps(selfPkg) ? `${selfPkg}` : ".";
+    const testSetup = `${withSelf}/config/test/setup.js`;
+
+    const mochaRequires = _.without(mochaOpts.require || [], tsNodeReg, sourceMapReg, testSetup);
+
+    if (this.appHasDevDeps("ts-node")) {
+      mochaRequires.push(tsNodeReg);
+    }
+
+    if (this.appHasDevDeps("source-map-support")) {
+      mochaRequires.push(sourceMapReg);
+    }
+
+    mochaRequires.push(testSetup);
+    mochaOpts.require = _.uniq(mochaRequires);
+    _.defaults(mochaOpts, { recursive: true });
+
+    if (this.saveAppPkgJson()) {
+      console.log(`INFO: updated mocha options in your package.json. Please commit it.`);
+    }
+  }
+
+  setupCoverage(): void {
+    const nyc = this.appPkg.nyc || {};
+
+    const nycConfigTs = "@istanbuljs/nyc-config-typescript";
+    const nycExtends = _.without(nyc.extends || [], nycConfigTs);
+    if (this.hasTypeScript) {
+      nycExtends.push(nycConfigTs);
+    }
+
+    nyc.extends = _.uniq(nycExtends);
+    _.defaults(nyc, {
+      all: true,
+      reporter: [],
+      exclude: [],
+      "check-coverage": true,
+      statements: 100,
+      branches: 100,
+      functions: 100,
+      lines: 100,
+      cache: !this.hasTypeScript
+    });
+    nyc.reporter = _.uniq(nyc.reporter.concat(["lcov", "text", "text-summary"]).sort());
+    nyc.exclude = _.uniq(
+      nyc.exclude.concat(["coverage", "*clap.js", "*clap.ts", "gulpfile.js", "dist", "test"]).sort()
+    );
+    this.appPkg.nyc = nyc;
+    if (this.saveAppPkgJson()) {
+      console.log("INFO: updated nyc config in your package.json. Please commit it.");
+    }
+  }
+}
+
+/**
+ * Make XClap build tasks
+ *
+ * @param options options
+ * @returns tasks
+ */
+function makeTasks(options: XarcModuleDevOptions): any {
```

Comment:
If you don't have a type to return, I would either let TS infer the type being returned by the implementation, or creating a new type.